Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/CommandLine.kt

```diff
@@ -16,97 +16,53 @@
-
-    @Argument(value = HEADER_FILTER_ADDITIONAL_SEARCH_PREFIX, shortName = "-hfasp",  valueDescription = "<file>", description = "header file to produce kotlin bindings for") 
-    var headerFilterPrefix: Array<String> = arrayOf()
-
-    @Argument(value = "-compilerOpts", shortName = "-copt", valueDescription = "<arg>", description = "additional compiler options", delimiter = " ")
-    var compilerOpts: Array<String> = arrayOf()
-
-    @Argument(value = "-linkerOpts", shortName = "-lopt", valueDescription = "<arg>", description = "additional linker options", delimiter = " ")
-    var linkerOpts: Array<String> = arrayOf()
-
-    @Argument(value = "-shims", description = "wrap bindings by a tracing layer") 
-    var shims: Boolean = false
-
-    @Argument(value = "-linker", valueDescription = "<file>", description = "use specified linker") 
-
-    var linker: String? = null
-    @Argument(value = "-cstubsname", valueDescription = "<name>", description = "provide a name for the generated c stubs file") 
-    var cstubsname: String? = null
+fun getCommonInteropArguments() = listOf(
+        OptionDescriptor(ArgType.Boolean(), "verbose", description = "Enable verbose logging output", defaultValue = "false"),
+        OptionDescriptor(ArgType.Choice(listOf("jvm", "native", "wasm")),
+                "flavor", description = "Interop target", defaultValue = "jvm"),
+        OptionDescriptor(ArgType.String(), "pkg", description = "place generated bindings to the package"),
+        OptionDescriptor(ArgType.String(), "generated", description = "place generated bindings to the directory",
+                defaultValue = System.getProperty("user.dir")),
+        OptionDescriptor(ArgType.String(), "libraryPath", description = "add a library search path",
+                isMultiple = true, delimiter = ","),
+        OptionDescriptor(ArgType.String(), "manifest", description = "library manifest addend"),
+        OptionDescriptor(ArgType.String(), "natives", description = "where to put the built native files",
+                defaultValue = System.getProperty("user.dir")),
+        OptionDescriptor(ArgType.String(), "staticLibrary", description = "embed static library to the result",
+                isMultiple = true, delimiter = ","),
+        OptionDescriptor(ArgType.String(), "temporaryFilesDir", description = "save temporary files to the given directory")
+    )
+
+fun getCInteropArguments(): List<OptionDescriptor> {
+    val options = listOf(
+            OptionDescriptor(ArgType.String(), "import", description = "a semicolon separated list of headers, prepended with the package name",
+                    isMultiple = true, delimiter = ";"),
+            OptionDescriptor(ArgType.String(), "target", description = "native target to compile to"),
+            OptionDescriptor(ArgType.String(), "def", description = "the library definition file"),
+            OptionDescriptor(ArgType.String(), "header", "hd", "header file to produce kotlin bindings for",
```

Comment:
Is it possible to keep the old short name for this option, `-h`? I would prefer to avoid altering command line interface.

Reply:
I don't know. In my opinion, it's common practice to use -h for getting help message. People expect such behaviour. I changed flags everywhere. @olonho ?

Reply:
`cinterop` CLI interface is unpopular but still public. Also there is at least `kotlin-multiplatform` Gradle plugin which is developed in other source repository.

Reply:
I understand, I don't know situation well, but getting error on asking help with common usage short name isn't very good too.
In MPP I changed to full name https://github.com/JetBrains/kotlin/pull/2117